res/controllers/Traktor-Kontrol-S4-MK2-hid-scripts.js

```diff
@@ -1088,6 +1137,29 @@ TraktorS4MK2.quantizeHandler = function(field) {
   TraktorS4MK2.controller.setOutput("[Master]", "!quantize", 0x7F * TraktorS4MK2.master_quantize, true);
 }
 
+TraktorS4MK2.snapHandler = function(field) {
+  if (field.value === 0) {
+    return;
+  }
+  library_maximized = engine.getValue("[Master]", "maximize_library");
+  engine.setValue("[Master]", "maximize_library", !library_maximized);
+}
+
+TraktorS4MK2.FXButtonHandler = function(field){
```

Comment:
In the future this should [work the same as the Components EffectUnit](https://mixxx.org/wiki/doku.php/standard_effects_mapping), but that will be quite a bit of work and I don't know if you can do that in the next few days for 2.1. For now, just getting the enable buttons and metaknobs working would be good. Are you still having trouble with the metaknobs?